### DISCUSSION TITLE
2.3.2 MYSQL not available if Playbook is not installed

### DISCUSSION THREAD

**Discussion by trb143** [Security-Onion-Solutions/securityonion#1765] [Unsupported Versions] [answered]
Using ISO 2.3.2 for a new install.

Selected Standalone but excluded Hive and Playbooks.

Install failed as it was trying to shutdown MySQL and the end of the the install.
Fleet also failed to install.

Neither MySQL nor Fleet dockers have been started.

Running so-mysql-start returns a fail.
Running so-fleet-setup returns errors as it cannot find the fleet and mysql contaners.

[fleet-setup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5469239/fleet-setup.log)
[mysql-start.txt](https://github.com/Security-Onion-Solutions/securityonion/files/5469240/mysql-start.txt)
[so-fleet-setup.txt](https://github.com/Security-Onion-Solutions/securityonion/files/5469241/so-fleet-setup.txt)
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/5469242/sosetup.log)

**Comment by stijnpieters** [comment]
Same here with distributed install. Both manager and sensor nodes "failed" install. They are both up and running, but no MySQL service. 
Fleet did install properly for me though, although it does not contain any hosts (presumably because MySQL doesn't work) and neither are there any plays in Playbook.

**Comment by trb143** [comment]
Done some more digging and spotted to follow.

so-docker-refresh downloaded a number of docker images event though we have a clean 2.3.2 install.

salt-call state.apply mysql queue=True  fails and gives no log messages

/opt/so/conf/mysql does not exist - is the fact that this is missing causing the services to load?
How can this be created in the correct manner?

running with debug option gives:

_[DEBUG   ] salt.crypt.get_rsa_key: Loading private key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/mysql/init.sls' using 'jinja' renderer: 0.2195606231689453
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/mysql/init.sls:





mysql_state_not_allowed:
  test.fail_without_changes:
    - name: mysql_state_not_allowed


[DEBUG   ] Results of YAML rendering: 
OrderedDict([('mysql_state_not_allowed', OrderedDict([('test.fail_without_changes', [OrderedDict([('name', 'mysql_state_not_allowed')])])]))])
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/mysql/init.sls' using 'yaml' renderer: 0.0003204345703125
[DEBUG   ] LazyLoaded config.option
[DEBUG   ] LazyLoaded test.fail_without_changes
[INFO    ] Running state [mysql_state_not_allowed] at time 16:35:13.186342
[INFO    ] Executing state test.fail_without_changes for [mysql_state_not_allowed]
[ERROR   ] Failure!
[INFO    ] Completed state [mysql_state_not_allowed] at time 16:35:13.187725 (duration_in_ms=1.384)
[DEBUG   ] File /var/cache/salt/minion/accumulator/139993143043856 does not exist, no need to cleanup_

**Reply by schmug** [reply]
What hardware are you running? Having the same trouble but have only been able to test ESXi.

**Reply by trb143** [reply]
Virtual Box - on laptop for test environment, Intel i7.
16 GB Ram - 7 CPU.

Looking at the init.sls code I think there is an issue with top_state but after that I am lost.

**Comment by trb143** [comment]
Think I have found it.
    {%- if FLEETMANAGER or FLEETNODE or PLAYBOOK != 0 %}
    - mysql
    {%- endif %}

If Playbbok is not wanted MYSQL will not be set up.

**Comment by m0duspwnens** [comment]
Thanks for the reports. https://github.com/Security-Onion-Solutions/securityonion/issues/1857 has been opened to track it.

**Comment by m0duspwnens** [comment, marked as answer]
A patch for this issue has been merged into dev (https://github.com/Security-Onion-Solutions/securityonion/pull/1863) and will be available in the next planned release.

Thanks again for the bug report.

**Reply by trb143** [reply]
Thanks for this.

I have manually made the change to me 2.3.2 machine.  
Is this the right way to do this?
How can I get the services to reset themselves so they start correctly?

**Reply by m0duspwnens** [reply]
Can you make this change as well?

https://github.com/Security-Onion-Solutions/securityonion/commit/5616aa6beb48c107178b8c0a56a517911afcecb2

That should fix the state top file and allow the mysql state to be applied which is what is causing issues for the fleet states.

You should now be able to run `salt-call state.show_top` and see the fleet and mysql states are going to be applied.

Once that change is made, run a `salt-call state.highstate` on the standalone node. This could resolve the issue, but there could be issues with the fleet packages that get generated during setup since fleet didn't get setup correctly at the start. If that is the case, we can either wait until the next ISO release to do a fresh install, or we can do a network install from the master branch and make these code changes prior to install.

**Comment by ldceven** [comment]
Hello, I am facing the same problem now, how can I solve it? Without waiting for a new ISO version
